This patch will allow you capture a picture using the volume keys.

Tested on: 1.2

Index: /usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
--- .orig/usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
+++ /usr/palm/applications/com.palm.app.camera/app/controllers/capture-assistant.js
@@ -34,6 +34,7 @@ CaptureAssistant = Class.create({
 			this.sublaunch = false;
 			this.defaultFilename = null;
 		}
+		this.volumeKeySubscription = undefined;		
 		
 	},
 
@@ -87,6 +88,15 @@ CaptureAssistant = Class.create({
 			var elemC = this.controller.get('sagar_console');
 			elemC.style.display = '';
 		}
+		// listen to volume key events
+		this.volumeKeySubscription = new Mojo.Service.Request(
+			'palm://com.palm.keys/audio', 
+			{
+				method: 'status',
+				parameters: {'subscribe': true},
+				onFailure: function() { Mojo.Log.error("Could not subscribe to volume key events"); },
+				onSuccess: this.handleVolumeKeys.bind(this), 
+			});
 		
 		llog("CaptureAssistant::setup() finished");
 	},
@@ -174,6 +184,11 @@ CaptureAssistant = Class.create({
 		}
 		
 		this.cameraControl.closeCamera();
+		
+		// clean up listener for volume keys
+		if(this.volumeKeySubscription) {
+			this.volumeKeySubscription.cancel();
+		}
 	},
 	
 	handleCommand: function(event){
@@ -210,6 +225,13 @@ CaptureAssistant = Class.create({
 
 	},
   
+	// capture on release of volume keys
+	handleVolumeKeys: function(payload) {
+		// capture when either volume up or down buttons are released
+		if(payload.state === 'up' && (payload.key === 'volume_up' || payload.key === 'volume_down')) {
+ 			this.cameraControl.capture();
+ 		}
+ 	},
 	
 	/**
 	 * Called by Mojo when the orientation of the device changes.
